daemon/analytics.c

```diff
@@ -0,0 +1,881 @@
+
+    BUFFER *bt = ap->both;
+
+    if (likely(ap->c)) {
+        buffer_strcat(bt, ",");
+    }
+
+    buffer_strcat(bt, "{");
+    buffer_strcat(bt, " \"plugin\": \"");
+    buffer_strcat(bt, col->plugin);
+    buffer_strcat(bt, "\", \"module\":\"");
+    buffer_strcat(bt, col->module);
+    buffer_strcat(bt, "\" }");
+
+    (ap->c)++;
+
+    return 0;
+}
+
+/*
+ * Create a JSON array of available collectors, same as in api/v1/info
+ */
+void analytics_collectors(void)
+{
+    RRDSET *st;
+    DICTIONARY *dict = dictionary_create(DICTIONARY_FLAG_SINGLE_THREADED);
+    char name[500];
+    BUFFER *bt = buffer_create(1000);
+
+    rrdset_foreach_read(st, localhost)
+    {
+        if (rrdset_is_available_for_viewers(st)) {
+            struct collector col = { .plugin = st->plugin_name ? st->plugin_name : "",
+                                     .module = st->module_name ? st->module_name : "" };
+            snprintfz(name, 499, "%s:%s", col.plugin, col.module);
+            dictionary_set(dict, name, &col, sizeof(struct collector));
+        }
+    }
+
+    struct array_printer ap;
+    ap.c = 0;
+    ap.both = bt;
+
+    dictionary_get_all(dict, collector_counter_callb, &ap);
+    dictionary_destroy(dict);
+
+    analytics_set_data(&analytics_data.netdata_collectors, (char *)buffer_tostring(ap.both));
+
+    {
+        char b[7];
+        snprintfz(b, 6, "%d", ap.c);
+        analytics_set_data(&analytics_data.netdata_collectors_count, b);
+    }
+
+    buffer_free(bt);
+}
+
+/*
+ * Run alarm-notify.sh script using the dump_methods parameter
+ * SEND_CUSTOM is always available
+ */
+void analytics_alarms_notifications(void)
+{
+    char *script;
+    script = mallocz(
+        sizeof(char) * (strlen(netdata_configured_primary_plugins_dir) + strlen("alarm-notify.sh dump_methods") + 2));
+    sprintf(script, "%s/%s", netdata_configured_primary_plugins_dir, "alarm-notify.sh");
+    if (unlikely(access(script, R_OK) != 0)) {
+        info("Alarm notify script %s not found.", script);
+        freez(script);
+        return;
+    }
+
+    strcat(script, " dump_methods");
+
+    pid_t command_pid;
+
+    info("Executing %s", script);
+
+    BUFFER *b = buffer_create(1000);
+    int cnt = 0;
+    FILE *fp = mypopen(script, &command_pid);
+    if (fp) {
+        char line[200 + 1];
+
+        while (fgets(line, 200, fp) != NULL) {
+            char *end = line;
+            while (*end && *end != '\n')
+                end++;
+            *end = '\0';
+
+            if (likely(cnt))
+                buffer_strcat(b, "|");
+
+            buffer_strcat(b, line);
+
+            cnt++;
+        }
+        mypclose(fp, command_pid);
+    }
+    freez(script);
+
+    analytics_set_data_str(&analytics_data.netdata_notification_methods, (char *)buffer_tostring(b));
+
+    buffer_free(b);
+}
+
+void analytics_charts(void)
+{
+    RRDSET *st;
+    int c = 0;
+    int show_archived = 0;
+    rrdset_foreach_read(st, localhost)
```

Comment:
show_archived should be removed

Reply:
Done (https://github.com/netdata/netdata/pull/10887/commits/67833e82af990879167d5ae97a6e75f8fef0070b)! Thanks Stelio.